lib/execution_engine2/utils/SDKMethodRunner.py

```diff
@@ -454,16 +457,65 @@ def update_job_status(self, job_id, status):
         if not (job_id and status):
             raise ValueError("Please provide both job_id and status")
 
-        with self.get_mongo_util().me_collection(self.config["mongo-jobs-collection"]):
+        job = self.get_mongo_util().get_job(job_id=job_id)
 
-            try:
-                job = Job.objects(id=job_id)[0]
-            except Exception:
-                raise ValueError(
-                    "Unable to find job:\nError:\n{}".format(traceback.format_exc())
-                )
+        if not job:
+            raise ValueError("Cannot find job with id: {}".format(job_id))
 
-            job.status = status
-            job.save()
+        job.status = status
+        job.save()
 
         return str(job.id)
+
+    def get_job_status(self, job_id):
+
+        returnVal = dict()
+
+        if not job_id:
+            raise ValueError("Please provide valid job_id")
+
+        job = self.get_mongo_util().get_job(job_id=job_id)
+
+        if not job:
+            raise ValueError("Cannot find job with id: {}".format(job_id))
+
+        returnVal['status'] = job.status
+
+        return returnVal
+
+    def finish_job(self, job_id, error_message=None):
+
+        if not job_id:
+            raise ValueError("Please provide valid job_id")
+
+        job = self.get_mongo_util().get_job(job_id=job_id)
+
+        if error_message:
+            job.errormsg = error_message
+            self.get_mongo_util().update_job_status(job_id=job_id, status=Status.error.value)
+        else:
+            self.get_mongo_util().update_job_status(job_id=job_id, status=Status.finished.value)
+
+        job.finished = datetime.utcnow()
+        job.save()
+
+    def start_job(self, job_id, skip_estimation=False):
+
+        if not job_id:
+            raise ValueError("Please provide valid job_id")
+
+        job = self.get_mongo_util().get_job(job_id=job_id)
+        job_status = job.status
+
+        if job_status == Status.estimating.value or skip_estimation:
+            # set job to running status
+            job.running = datetime.utcnow()
+            self.get_mongo_util().update_job_status(job_id=job_id, status=Status.running.value)
+        elif job_status in [Status.created.value, Status.queued.value]:
```

Comment:
Very good. I think we should first check the elif condition first, since that would follow the logic structure of

runjob -> start_job -> start_job -> finish_job
Created -> Estimating -> Running -> Finished

I think it makes the code a little easier to follow to do it that way. We should probably document the flow of this in the comments for this function too

Reply:
👍